### PR TITLE
Adding Multiple Date Picker

### PR DESCRIPTION
**Pull request by jasonvalenzuela**
Story link: https://www.pivotaltracker.com/story/show/166741446

Adds multiple date picker to activity form.

### CODE REVIEW

File: src/components/MultipleDatePicker.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    this.show = true
+  }
+
+  @action.bound closeCalendar = (event) => {
+    event.preventDefault()
+    this.show = false
+  }
+
+  @action.bound handleDayClick = (day, { selected }) => {
+    const { value } = this.props
```
Comment (ryan-peterson-emberex) [inline src/components/MultipleDatePicker.js]: Props shouldn't be changed directly like this. Have the parent component pass an onChange-like listener (ex: `onDayClicked`) and call that function with the new value.

File: src/components/EditActivityForm.js [hunk 2]
```diff
@@ -62,23 +63,27 @@ class EditActivityForm extends Component {
...
+        return;
+      }
+      if (this.events.every(event => event.eventTime.getTime() !== date.getTime())) {
+        this.events.push({
+          id: `event-${tempIdCounter++}`,
+          eventTime: date
+        });
+      }
+    })
+    this.eventTimes.splice(0, this.eventTimes.length)
```
Comment (ryan-peterson-emberex) [inline src/components/EditActivityForm.js]: Call me opinionated, but `this.eventTimes = [];` looks cleaner. This also probably breaks some React state rule (I'm no expert though).

Comment (jasonvalenzuela) [reply src/components/EditActivityForm.js]: Definitely does! Not sure why I cleared the array with `splice()` before.

File: src/components/MultipleDatePicker.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+import { observable, action } from 'mobx';
+import { observer } from 'mobx-react';
+import { Calendar } from './Icons';
+
+@observer
+class MultipleDatePicker extends Component {
+  @observable show = false
+  @observable current = new Date()
+
+  @action.bound YearMonthForm = ({ date, localeUtils }) => {
```
Comment (jmeyers91) [inline src/components/MultipleDatePicker.js]: This doesn't need the `@action.bound` decorator. The action decorator is used to tell mobx that a function might change observable values. The `handleChange` function for example should probably be an action.

File: src/components/MultipleDatePicker.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            <option key={year} value={year}>
+              {year}
+            </option>
+          ))}
+        </select>
+      </div>
+    )
+  }
+
+  @action.bound showCalendar = (event) => {
```
Comment (jmeyers91) [inline src/components/MultipleDatePicker.js]: No need for arrow function properties when using `action.bound`. The `bound` part auto binds the action to the component class.

**Pull request merged**